modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java

```diff
@@ -83,7 +85,10 @@ public static void addAcceleratorsIntoScene(ObservableList<MenuItem> items, Node
         // 2. Removing accelerators when Control is removed from Scene
         // Remove previously added listener if any
         if (sceneChangeListenerMap.containsKey(anchor)) {
-            anchor.sceneProperty().removeListener(sceneChangeListenerMap.get(anchor));
+            ChangeListener<Scene> listener = sceneChangeListenerMap.get(anchor).get();
```

Comment:
It seems unusual to check for the existence of a weak key (containsKey), and then just assume it still exists (get). You should probably get() the value directly without checking containsKey() first, and then check whether the returned value is null.